aggregator_core/src/datastore.rs

```diff
@@ -209,6 +216,7 @@ impl<C: Clock> Datastore<C> {
         for<'a> F:
             Fn(&'a Transaction<C>) -> Pin<Box<dyn Future<Output = Result<T, Error>> + Send + 'a>>,
     {
+        let mut retry_count = 0;
```

Comment:
outside the scope of this PR: given that we've tripped over an infinite-retry on transactions at least once, I think we should eventually implement a maximum-retry-count to act as a circuit breaker. However, that can wait until at least we know what realistic retry counts look like, perhaps based on feedback from this metric once it's deployed. We should probably set our maximum retry count to at least an order of magnitude higher than that (and yell/alert as loudly as possible if we start to encounter it, since it either indicates that our maximum retry count is too low or that we've found a serious bug).

Reply:
This idea is captured in https://github.com/divviup/janus/issues/2124.

Reply:
Agreed. I was hoping that [introducing `backoff` to this](https://github.com/divviup/janus/pull/2381) logic would handle cancellation for us, but I found exponential backoff to be regressive.